### ISSUE TITLE
Hi!

### ISSUE THREAD

**Issue by bartchwalek** [Roger-random/mitutoyo#1]
Hopefully you read this.

So I browsed the whole web and found your version of the implementation for the mitutoyo digimatic the best... However I think we are missing data for a better and more complete implementation. I was wondering, since I did it for my dial indicator, if there was some commands that could be sent as well to the unit... but anyways, my main topic here is 

1. I have ported the implementation to ESP8266, you should know what this mcu is, in a few words its a sub 10$ kinda IOT module, I am using the nodeMCU v3 chinese version... 
2. It is implemented in Python, microPython (upython)
3. It connects to my local AP and
4. It multicasts the value on the network, so on any machine I can listen to my digital dial indicator value....

I think this is pretty awesome and in all, the dial indicator I bought of ebay cost me about 30$ shipped since it was broken (the lm7805 was burnt and the guy did not have the power supply... its a vintage model, but rugged... series543-513... didn't find the manual yet... but it is very early digimatic.) The module esp8266 about 5$ china or 10$ amazon... transistor + resistors and some cable and as well another LM7805 to power the esp... 2$... a 12v 2A power supply about 8$... in all, for around 50$ I have a precise wifi hooked multicasting digital dial indicator! wow

So, I use udp multicasting to reduce lag and the ESP has only 4mb flash... so although it is possible, I didn't want to implement a whole http server + app... I will make a local interface that shows the value on graph etc..

I will publish the project sometime soon with a tutorial most probably .

If you have any suggestions, comments or ideas, please share!

Sorry I am using the issues in github, I didnt figure out how to contact u directly (ugh)

**Comment by Roger-random** [comment]
Hello, welcome! I'm glad you found my project.

In case it wasn't obvious, my project was not focused on "better" or "complete". It was very specifically a "get it barely working then move on" kind of project. And I shared it in its very rough and ugly state because I have no shame about such things.

The documentation that came with my Mitutoyo indicator had no information on sending commands to the unit. I thought it might be nice to do things like setting inch/mm or zero the display, but it was not critical to my project so I moved on.

I'm familiar with the ESP8266 however my particular project didn't need wireless so I used a classic Arduino.

For my wireless data transmission projects, I've been using the MQTT protocol. MQTT is popular and almost every development environment/hardware has a MQTT library. Certainly the ESP8266 does! Once data gets to a MQTT broker it's easy to perform further processing with something like Node-RED, or store the measurement into a database like InfluxDB. Caveat: I haven't worked on a project where low lag is required, so I don't know how they measure up.

Once you publish your project, I'm happy to add a link on my README file.
